RecoEgamma/EgammaIsolationAlgos/interface/EgammaL1TkIsolation.h

```diff
@@ -0,0 +1,64 @@
+#ifndef RECOEGAMMA_EGAMMAISOLATIONALGOS_EGAMMAL1TKISOLATION_H
+#define RECOEGAMMA_EGAMMAISOLATIONALGOS_EGAMMAL1TKISOLATION_H
+
+#include "DataFormats/L1TrackTrigger/interface/L1Track.h"
+#include "DataFormats/TrackReco/interface/TrackBase.h"
+#include "DataFormats/TrackReco/interface/TrackFwd.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+
+//author S. Harper (RAL/CERN)
+//based on the work of Swagata Mukherjee and Giulia Sorrentino
+
+class EgammaL1TkIsolation {
+public:
+  explicit EgammaL1TkIsolation(const edm::ParameterSet& para);
+  EgammaL1TkIsolation(const EgammaL1TkIsolation&) = default;
+  ~EgammaL1TkIsolation() = default;
+  EgammaL1TkIsolation& operator=(const EgammaL1TkIsolation&) = default;
```

Comment:
it's not totally clear why this is needed?

Reply:
I like to telegraph my intentions. It also shows at a glance that the author likely has some clue about the pitfalls of c++ copies and didnt simply lack that knowledge. There is far to much copy unsafe code in CMSSW.

Reply:
And on 4.26, urgh. I do like to group methods of sub classes and not split them but I guess the consequences of 4.26 mean I can no longer do that. Fine.

Reply:
> I like to telegraph my intentions.

> 4.13: For a class, definition of any of the following requires definition of all five: destructor, copy constructor, copy assignment operator, move constructor, and move assignment operator. (*)

If I interpret your comment and 4.13 at face value, you should then also then explicitly define (as defaults) the move constructor and move assignment operators, which are currently implicit, or make them all implicit.

Reply:
Well its a rather pedantic interpretation of 4.13 isnt it?  4.13 clearly is intended to avoid the situation where a custom destructor is needed and thus any attempt to move, copy or assign the class will almost certainty cause problems if the other 4 arent defined. Given that adding or removing these lines doesnt functionally change anything, does it really count as a definition.

Reply:
fine.

Reply:
Quoting from [C++ Core Guidelines C.21](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rc-five)
> Declaring any copy/move/destructor function, even as `=default` or `=delete`, will suppress the implicit declaration of a move constructor and move assignment operator.

Therefore it would be better to define all five or none of them.

Reply:
I already removed them

Reply:
and thank you for the clarification here, I agree my original proposal was incorrect.

Reply:
Thank you Matti, I learned something new here as well.